File: solutions/ruby/space-age/space_age.rb
class SpaceAge
  def initialize(seconds)
    @seconds = seconds
  end

  def on_earth
    with_rate(1.0)
  end

  def on_mercury
    with_rate(0.2408467)
  end

  def on_venus
    with_rate(0.61519726)
  end

  def on_mars
    with_rate(1.8808158)
  end

  def on_jupiter
    with_rate(11.862615)
  end

  def on_saturn
    with_rate(29.447498)
  end

  def on_uranus
    with_rate(84.016846)
  end

  def on_neptune
    with_rate(164.79132)
  end

private
  def with_rate(n)
    (@seconds * (1 / n)) / 31557600
  end
end

File: solutions/python/space-age/space_age.py
class SpaceAge:
    def __init__(self, seconds):
        self.seconds = seconds

    def on_earth(self):
        return self._with_rate(1.0)

    def on_mercury(self):
        return self._with_rate(0.2408467)

    def on_venus(self):
        return self._with_rate(0.61519726)

    def on_mars(self):
        return self._with_rate(1.8808158)

    def on_jupiter(self):
        return self._with_rate(11.862615)

    def on_saturn(self):
        return self._with_rate(29.447498)

    def on_uranus(self):
        return self._with_rate(84.016846)

    def on_neptune(self):
        return self._with_rate(164.79132)

    def _with_rate(self, n):
        return round(self.seconds * (1 / n) / 31557600, 2)